modules/saml/src/Auth/Source/SP.php

```diff
@@ -560,17 +558,12 @@ private function startSSO2(Configuration $idpMetadata, array $state): void
 
         if (isset($state['saml:NameIDPolicy'])) {
             $policy = null;
```

Comment:
This should now default to `[]` I think... Psalm is now complaining about passing a possible null-value to `setNameIdPolicy()` on line 567

Reply:
I've made it even simpler now.

Reply:
Excellent! I think it's ready to merge